api/src/main/java/org/openmrs/api/impl/UserServiceImpl.java

```diff
@@ -712,9 +726,35 @@ public User getUserByEmail(String email) throws APIException {
 	 * 
 	 */
 	@Override
-	@Transactional(readOnly = true)
-	public User getUserByActivationKey(String activationKey) throws APIException {	
-		return dao.getUserByActivationKey(activationKey);
+	public User getUserByActivationKey(String activationKey) {
+		LoginCredential loginCred = dao.getLoginCredentialByActivationKey(activationKey);
+		String[] credTokens = loginCred.getActivationKey().split(":");
+
+		if(loginCred != null  && (System.currentTimeMillis() <= Long.parseLong(credTokens[1]) )) {
```

Comment:
Don't do this in the dao, whether the token is expired or not, the dao should return the token, it's not its business to check for expired tokens